stock_scanner_web/controllers/main.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Angel Moya <http://angelmoya.es>
+# Copyright 2016 Eficent Business and IT Consulting Services, S.L.
+# <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import http
+from openerp.http import request
+from openerp import _
+
+
+def allowed_hardware(user, t_num):
+    allowed = False
+    for hardware in user.scanner_hardware_ids:
+        if t_num == hardware.code:
+            allowed = True
+    return allowed
+
+
+class ScannerWeb(http.Controller):
+    @http.route([
+        '/scanner_call',
+        '/scanner_call/<string:terminal_number>',
+        '/scanner_call/<string:terminal_number>/<string:action>',
+        '/scanner_call/<string:terminal_number>/<string:action>/'
+        '<string:message>',
+    ], website=True, auth='user')
+    def scanner_call(self,
+                     terminal_number='',
+                     action='',
+                     message=False,
+                     type='http',
+                     auth='public',
+                     website=True):
+        values = {}
+        if message == 'False':
+            message = False
+        try:
```

Comment:
This try/except is quite long & I am unable to determine what line(s) it is actually guarding. It would be better to immediately perform the action that could generate the exception & return. It would also reduce the nesting